app/components/Login/Login.jsx

```diff
@@ -40,14 +47,51 @@ export default class Login extends React.Component {
         this.setState({ show: true});
     }
 
+    validateUsernamePassword(e) {
+        if (e.keyCode === 13) {
+            if (!window.localStorage.getItem("vaultUrl")) {
+                this.setState({errorMessage: "No Vault url specified.  Click the gear to edit your Vault url"});
+                return;
+            }
+            if (!this.state.username) {
+                this.setState({errorMessage: "No username provided."});
+                return;
+            }
+
+            if (!this.state.password) {
+                this.setState({errorMessage: "No password provided."});
+                return;
+            }
+            axios.post('/login', {
+                "VaultUrl": window.localStorage.getItem("vaultUrl"),
+                "Creds": {
+                    "Type": this.state.loginMethodType,
+                    "Username": this.state.username,
+                    "password": this.state.password
+                }
+            })
+            .then((resp) => {
+                //DJ fill in the blanks here
```

Comment:
DJ you'll need to handle this part - and on the server as well.  Right now, anything besides the Github auth method throws a 400 bad request error server side.